### PR TITLE
cpp-dump: add recipe

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **cpp-dump/0.5.0**

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/cpp-dump/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+required_conan_version = ">=1.52.0"
+
+
+class CppDumpConan(ConanFile):
+    name = "cpp-dump"
+    description = "An all-round dump function library for C++ that can print any type, even user-defined types."
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/philip82148/cpp-dump"
+    topics = ("debug", "dump" "header-only")
```
Comment (valgur) [inline recipes/cpp-dump/all/conanfile.py:19]:
```suggestion
    topics = ("debug", "dump", "header-only")
```

File: recipes/cpp-dump/all/test_package/test_package.cpp [hunk 2]
```diff
@@ -0,0 +1,14 @@
+#include <iostream>
+#include <ostream>
+#include <vector>
+
+#include "dump.hpp"
```
Comment (valgur) [inline recipes/cpp-dump/all/test_package/test_package.cpp:5]:
```suggestion
#include <dump.hpp>

#include <iostream>
#include <ostream>
#include <vector>
```
To verify that no standard library includes are missing in the library.

**Review by valgur — CHANGES_REQUESTED** [review]
LGTM, except for the one typo.

**Review by RubenRBS — APPROVED** [review]
Thanks!

**Review by jwillikers — APPROVED** [review]

**Review by czoido — APPROVED** [review]

**Pull request merged**